### DISCUSSION TITLE
Feature request:  Primary vs Fork indicator

### DISCUSSION THREAD

**Discussion by hrieke** [npm/feedback#344] [Website] [answered]
Would it be possible to add a checkmark, golden wombat, or some form of indication on the page that the author of the package is also the author of the code?
Thanks,

**Comment by ljharb** [comment]
How could that be known? There's no guaranteed link between an npm user and a github user, and packages aren't required to have any repo at all, let alone a github one.

**Reply by hrieke** [reply]
I'm sure that there are ways to verify if the code repo is a primary or a fork.  Github shows that relationship clearly.
And since Github owns NPM, they might come up with something.
To the broader statements in your comment:
* There's no guaranteed link between an npm user and a github user - Correct, but all the more reason to think about the code and whom it is coming from.  If I fork a random JS code base because the original author has no interest in publishing it, does this make me the authoritative owner of the package, specially in the condition that I've placed it here as a convenance for myself?
* packages aren't required to have any repo at all - Don't you just hate that?  I do.  Makes code audits a pain.
* let alone a Github one - Non github codebases (Gitlab for example) also show Fork status.  Clearly if someone forks a project from Github to Gitlab the Fork status is broken.

In any case I really don't want NPM to become another NuGet, where random users posted half of MS' dlls...

**Reply by ljharb** [reply]
Yes, it does make you the owner of the package - the forker owns the fork.

Code audits are always a pain, because any package with a build process won't have the outputs in git and won't necessarily have the inputs in the package.

**Reply by hrieke** [reply]
I feel otherwise, and a status indication / verification of sorts would be a good thing.
As someone who performs about 6 audits a month, I will say anything that makes my life easier is a good thing.

**Comment by MylesBorins** [comment, marked as answer]
@hrieke we are definitely exploring the ways we can better utilize npm + GitHub data! I'm not sure that showing if something is a "fork" makes the most sense, but we are looking into ways that we can better state that a specific package is connected to a specific repo.

There is no way to guarantee that a tag of a release in a repository will be byte to byte equivalent to a release on npm, but we have also been thinking through the ways we can verify that a specific tag on GitHub resulted in a specific release on npm.

**Reply by hrieke** [reply]
Thank you.  
Anything to keep NPM from become another NuGet blackhole of junk is a good step forward in my opinion.

The minimum requirements in my mind are the Who, What, Where, Version, License.
If you can answer those five questions then the component should be marked as verified.

Just to give you an idea of what a mess NuGet has become, I have my casebook example, Microsoft DLLs being published by a third party as an ease for their own use:
Microsoft.WindowsAzure.SDK
https://www.nuget.org/packages/Microsoft.WindowsAzure.SDK

1.5 million downloads. 🤯
